### PR TITLE
make bucket prefix for logical backup configurable

### PR DESCRIPTION
**Pull request by FxKu**
In order to configure separate S3 lifecycle rules for basebackups and logical backups living in the same bucket, we have to make the logical backup prefix configurable. Unlike Spilo, this approach works in the logical backup script.

There's one more problem though: The operator only diffs logical backup jobs on their image and schedule. Thus, existing jobs will not be updated when changing the prefix. This problem has also been described in issue #768.

Therefore, I have extended the diff to also compare containers for which I have reused the comparison method of the statefulset pod template. It's not super clean - there's a check for lazy update which does not apply for the logical backup cronjob. The word `statefulset` must now be passed as part of the description to not produce confusing logs for cron job diffs.

Because the cronjob diff function now calls `compareContainers` - a `cluster` function - the function needs a cluster receiver, too. I moved it from k8sutil to cluster package and renamed it to align it with other diff functions.

With this PR I have also move the actual logical backup code outside of the `docker` folder, because it should exist on the same level as e.g. the `ui` (and future `pooler` folder). Adjusted the build scripts accordingly.

### CODE REVIEW

**Review by hughcapet — COMMENTED** [review]
Will it make sense to also update the [e2e test](https://github.com/zalando/postgres-operator/blob/lb-bucket-prefix/e2e/tests/test_e2e.py#L1108) ?

File: docs/reference/operator_parameters.md [hunk 1]
```diff
@@ -845,6 +845,9 @@ grouped under the `logical_backup` key.
   S3 bucket to store backup results. The bucket has to be present and
   accessible by Postgres pods. Default: empty.
 
+* **logical_backup_s3_bucket_prefix**
```
[the remark this answers is not in the document]
Comment (FxKu) [reply docs/reference/operator_parameters.md:848]: I've added the tailing / to the logical backup script. Even if users leave the prefix blank // will be treated as /. So it's not needed to specify it.

**Comment by hughcapet** [comment]
👍

**Comment by FxKu** [comment]
:+1:

**Pull request merged**